common/src/main/java/org/opensearch/ml/common/MLCommonsClassLoader.java

```diff
@@ -277,7 +277,7 @@ private static <T, S> S init(Map<T, Class<?>> map, T type,
             Throwable cause = e.getCause();
             if (cause instanceof MLException) {
                 throw (MLException)cause;
-	    } else if (cause instanceof IllegalArgumentException) {
```

Comment:
The two exceptions can be merged together, e.g. `cause instanceof MLException || cause instanceof IllegalArgumentException`.

Reply:
**MLException** and **IllegalArgumentException** are both runtime exceptions. We treat them separately so that we do not need to explicitly declare **throw Throwable** in function signature.